subsystem/interlocks.py

```diff
@@ -1,60 +1,157 @@
 
-        interlock_labels = [
-            "Vacuum", "Water", "Door", "Timer", "Oil High",
-            "Oil Low", "E-stop Ext", "E-stop Int", "G9SP Active"
-        ]
+        self.parent.grid_rowconfigure(0, weight=1)
+        self.parent.grid_columnconfigure(0, weight=1)
+
+        self.interlocks_frame.grid_rowconfigure(0, weight=1)
+        self.interlocks_frame.grid_columnconfigure(0, weight=1)
+        self.interlocks_frame.grid(row=0, column=0, sticky='nsew')
+
+        interlocks_frame = tk.Frame(self.interlocks_frame, highlightbackground="black")
+        interlocks_frame.grid(row=0, column=0, padx=0, pady=0, sticky="nsew")
+
+        num_columns = 22  
+        for i in range(num_columns):
+            interlocks_frame.grid_columnconfigure(i, weight=1)
+
         self.indicators = {
-            'active': tk.PhotoImage(file=resource_path("media/off_orange.png")),
-            'inactive': tk.PhotoImage(file=resource_path("media/on.png"))
-        }
-
-        for label in interlock_labels:
-            frame = tk.Frame(self.interlocks_frame)
-            frame.pack(side=tk.LEFT, expand=True, padx=5)
-
-            lbl = tk.Label(frame, text=label, font=("Helvetica", 8))
-            lbl.pack(side=tk.LEFT)
-            status = self.interlock_status[label]
-            indicator = tk.Label(frame, image=self.indicators['active'] if status else self.indicators['inactive'])
-            indicator.pack(side=tk.RIGHT, pady=1)
-            frame.indicator = indicator  # Store reference to the indicator for future updates
-
-    def update_interlock(self, name, status):
-        if name in self.parent.children:
-            frame = self.parent.children[name]
-            indicator = frame.indicator
-            new_image = self.indicators['active'] if status else self.indicators['inactive']
-            indicator.config(image=new_image)
-            indicator.image = new_image  # Keep a reference
-
-    def update_pressure_dependent_locks(self, pressure):
-        # Disable the Vacuum lock if pressure is below 2 mbar
-        self.update_interlock("Vacuum", pressure >= 2)
\ No newline at end of file
+            'Door': None, 'Water': None, 'Vac Power': None, 
+            'Vac Pressure': None, 'Low Oil': None, 'High Oil': None, 
+            'E-STOP Int': None, 'E-STOP Ext' : None, 'All Interlocks': None, 
+            'G9 Output': None, 'HVOLT ON': None
+                      }
+        
+        # Makes all the inticators and labels
+        for i, (k,v) in enumerate(self.indicators.items()):
+            tk.Label(interlocks_frame, text=f"{k}", anchor="center").grid(row=0, column=i*2, sticky='ew')
+            canvas, oval_id = create_indicator_circle(interlocks_frame, 'red')
+            canvas.grid(row=0, column=i*2+1, sticky='nsew')
+            self.indicators[k] = (canvas, oval_id)
+
+    # logging the history of updates
+    def update_interlock(self, name, safety, data):
+        # means good
+        if safety & data == 1:
```

Comment:
I think there's an issue with this logic here. The `==` operator has higher precedence than `&` so this is evaluated as `safety & (data == 1)` not `(safety & data) == 1` (which is how I think it should be)